Applications/DataExplorer/DataView/MeshElementRemovalDialog.cpp

```diff
@@ -202,24 +262,48 @@ void MeshElementRemovalDialog::on_meshNameComboBox_currentIndexChanged(int idx)
-    this->materialListWidget->clearSelection();
-    if (this->boundingBoxCheckBox->isChecked()) this->on_boundingBoxCheckBox_toggled(true);
-    auto mesh = _project.getMesh(meshNameComboBox->currentText().toStdString());
-    if (mesh->getProperties().existsPropertyVector<int>("MaterialIDs"))
-    {
-        auto const* const materialIds =
-            mesh->getProperties().getPropertyVector<int>("MaterialIDs");
-        if (materialIds->size() != mesh->getNumberOfElements())
-        {
-            ERR ("Incorrect mesh structure: Number of Material IDs does not match number of mesh elements.");
-            OGSError::box("Incorrect mesh structure detected.\n Number of Material IDs does not match number of mesh elements");
-            QMetaObject::invokeMethod(this, "close", Qt::QueuedConnection);
-        }
+    this->scalarArrayComboBox->clear();
+    this->outsideScalarMinEdit->setText("");
+    this->outsideScalarMaxEdit->setText("");
+    this->insideScalarMinEdit->setText("");
+    this->insideScalarMaxEdit->setText("");
+    if (this->scalarArrayCheckBox->isChecked())
+        on_scalarArrayCheckBox_toggled(true);
+    if (this->boundingBoxCheckBox->isChecked())
+        on_boundingBoxCheckBox_toggled(true);
+}
+
+void MeshElementRemovalDialog::on_scalarArrayComboBox_currentIndexChanged(int idx)
+{
+    Q_UNUSED(idx);
+    MeshLib::Mesh const* const mesh =
+        _project.getMesh(meshNameComboBox->currentText().toStdString());
+    MeshLib::Properties const& properties = mesh->getProperties();
 
-        materialIDCheckBox->setEnabled(true);
-        if (materialIDCheckBox->isChecked())
-            on_materialIDCheckBox_toggled(true);
+    std::string const vec_name(scalarArrayComboBox->currentText().toStdString());
+    if (vec_name == "")
+        return;
+
+    if (properties.existsPropertyVector<int>(vec_name))
+    {
+        MeshLib::PropertyVector<int> const& vec =
+            *properties.getPropertyVector<int>(vec_name);
+        auto min = std::min_element(vec.cbegin(), vec.cend());
+        auto max = std::max_element(vec.cbegin(), vec.cend());
+        this->outsideScalarMinEdit->setText(QString::number(*min));
+        this->outsideScalarMaxEdit->setText(QString::number(*max));
+        this->insideScalarMinEdit->setText(QString::number(*min));
+        this->insideScalarMaxEdit->setText(QString::number(*max));
+    }
+    else if (properties.existsPropertyVector<double>(vec_name))
+    {
+        MeshLib::PropertyVector<double> const& vec =
+            *properties.getPropertyVector<double>(vec_name);
+        auto min = std::min_element(vec.cbegin(), vec.cend());
+        auto max = std::max_element(vec.cbegin(), vec.cend());
+        this->outsideScalarMinEdit->setText(QString::number(*min));
+        this->outsideScalarMaxEdit->setText(QString::number(*max));
+        this->insideScalarMinEdit->setText(QString::number(*min));
+        this->insideScalarMaxEdit->setText(QString::number(*max));
```

Comment:
The `int` and `double` versions are equal up to the type. A lambda would remove the code duplication easily.
```c++
template <typename T> auto do_something = [&]() {
   MeshLib::PropertyVector<T> ...
}
```